File: javascript/get-token.js
const request = require('request-promise');

const clientid = process.argv[2];
const clientsecret = process.argv[3];
const username = process.argv[4];
const password = process.argv[5];

auth = 'Basic ' + new Buffer(clientid + ':' + clientsecret).toString('base64');
request(
    { 
        method: 'POST',
        uri: 'https://sandbox.xill.io/oauth/token',
        headers: {
            'Authorization': auth,
            'Content-Type': 'application/x-www-form-urlencoded;charset=UTF-8'
        },
        json: true,
        body: 'grant_type=password&username=' + username + '&password=' + password
    })
    .then((body) => {
        console.log(JSON.stringify(body, ' ', 2));
    });
    

File: java/src/main/java/com/xillio/apiexamples/GetToken.java
package com.xillio.apiexamples;

import org.json.JSONObject;

import java.util.Base64;


public class GetToken {
    public static void main(String args[]) {
        String clientId = args[1];
        String clientSecret = args[2];
        String username = args[3];
        String password = args[4];
        byte[] bytesEncoded = Base64.getEncoder().encode((clientId + ":" + clientSecret).getBytes());
        String auth = "Basic " + new String(bytesEncoded);
        JSONObject oauth = Utils.postData("https://sandbox.xill.io/oauth/token", "grant_type=password&username=" + username + "&password=" + password, auth);

        System.out.println(oauth.toString(2));
    }
}